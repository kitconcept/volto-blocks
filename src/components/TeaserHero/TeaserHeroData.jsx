import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl } from 'react-intl';
import { Form, Grid, Segment } from 'semantic-ui-react';
import TeaserData from '../TeaserGrid/TeaserData';
import TeaserHeroVariations from './TeaserHeroVariations';

const TeaserHeroData = ({
  data,
  tile,
  onChangeTile,
  required = false,
  intl,
}) => {
  return (
    <>
      <TeaserData data={data} tile={tile} onChangeTile={onChangeTile} />
      <Segment className="form sidebar-image-data">
        <Form.Field inline>
          <Grid>
            <Grid.Row>
              <Grid.Column width="4">
                <div className="wrapper">
                  <label htmlFor="field-align">
                    <FormattedMessage
                      id="Variation"
                      defaultMessage="Variation"
                    />
                  </label>
                </div>
              </Grid.Column>
              <Grid.Column width="8" className="align-tools">
                <TeaserHeroVariations
                  variation={data.variation}
                  onChangeTile={onChangeTile}
                  data={data}
                  tile={tile}
                />
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Form.Field>
      </Segment>
    </>
  );
};

TeaserHeroData.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  tile: PropTypes.string.isRequired,
  onChangeTile: PropTypes.func.isRequired,
};

export default injectIntl(TeaserHeroData);
