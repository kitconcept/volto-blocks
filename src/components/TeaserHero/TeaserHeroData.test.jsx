import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import TeaserHeroData from './TeaserHeroData';

const mockStore = configureStore();

test('renders an Listing Data Sidebar component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <TeaserHeroData
        id="1c58b5a2-26fb-4112-832b-fad88af1ae0e"
        data={{
          '@type': 'teaserHero',
          href: '/de/a-page/news1',
          variation: 'left',
          padded: true,
        }}
        block="1c58b5a2-26fb-4112-832b-fad88af1ae0e"
        pathname="/news"
        onChangeBlock={() => {}}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
