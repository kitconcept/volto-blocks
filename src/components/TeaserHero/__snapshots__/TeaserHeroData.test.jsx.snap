// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an Listing Data Sidebar component 1`] = `
Array [
  <div
    className="ui segment form sidebar-image-data"
  >
    <div
      className="inline field text"
      id="field-source-0"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-source-0"
              >
                Source
              </label>
            </div>
          </div>
          <div
            className="eight wide column"
          >
            <div
              className="ui icon input"
            >
              <input
                id="field-source-0"
                maxLength={null}
                minLength={null}
                name="source-0"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                type="text"
                value="/de/a-page/news1"
              />
              <i
                aria-hidden="true"
                attributes={
                  Object {
                    "viewBox": "",
                    "xmlns": "",
                  }
                }
                className="icon"
                onClick={[Function]}
              />
            </div>
            <button
              onClick={[Function]}
            >
              <svg
                className="icon"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
                onClick={null}
                style={
                  Object {
                    "fill": "currentColor",
                    "height": "18px",
                    "width": "auto",
                  }
                }
                viewBox=""
                xmlns=""
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="inline field text"
      id="field-title-0"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-title-0"
              >
                Title
              </label>
            </div>
          </div>
          <div
            className="eight wide column"
          >
            <div
              className="ui input"
            >
              <input
                id="field-title-0"
                maxLength={null}
                minLength={null}
                name="title-0"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="inline field text"
      id="field-description-0"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-description-0"
              >
                Description
              </label>
            </div>
          </div>
          <div
            className="eight wide column"
          >
            <div
              className="ui input"
            >
              <input
                id="field-description-0"
                maxLength={null}
                minLength={null}
                name="description-0"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="inline field text"
      id="field-source-preview-image"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-source-preview-image"
              >
                Source image
              </label>
            </div>
          </div>
          <div
            className="eight wide column"
          >
            <div
              className="ui icon input"
            >
              <input
                id="field-source-preview-image"
                maxLength={null}
                minLength={null}
                name="source-preview-image"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                type="text"
                value=""
              />
              <i
                aria-hidden="true"
                attributes={
                  Object {
                    "viewBox": "",
                    "xmlns": "",
                  }
                }
                className="icon"
                onClick={[Function]}
              />
            </div>
            <button
              onClick={[Function]}
            >
              <svg
                className="icon"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
                onClick={null}
                style={
                  Object {
                    "fill": "currentColor",
                    "height": "18px",
                    "width": "auto",
                  }
                }
                viewBox=""
                xmlns=""
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="inline field"
      id="field-openLinkInNewTab"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <div
                className="ui checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  name="field-openLinkInNewTab"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label
                  htmlFor="field-openLinkInNewTab"
                >
                  Open in a new tab
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui segment form sidebar-image-data"
  >
    <div
      className="inline field"
    >
      <div
        className="ui grid"
      >
        <div
          className="row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-align"
              >
                Variation
              </label>
            </div>
          </div>
          <div
            className="eight wide column align-tools"
          >
            <div>
              <div
                className="ui buttons"
              >
                <button
                  aria-label="Top"
                  className="ui basic icon button"
                  onClick={[Function]}
                >
                  <svg
                    className="icon"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": undefined,
                      }
                    }
                    onClick={null}
                    style={
                      Object {
                        "fill": "currentColor",
                        "height": "24px",
                        "width": "auto",
                      }
                    }
                    viewBox=""
                    xmlns=""
                  />
                </button>
              </div>
              <div
                className="ui buttons"
              >
                <button
                  aria-label="Left"
                  className="ui active basic icon button"
                  onClick={[Function]}
                >
                  <svg
                    className="icon"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": undefined,
                      }
                    }
                    onClick={null}
                    style={
                      Object {
                        "fill": "currentColor",
                        "height": "24px",
                        "width": "auto",
                      }
                    }
                    viewBox=""
                    xmlns=""
                  />
                </button>
              </div>
              <div
                className="ui buttons"
              >
                <button
                  aria-label="Right"
                  className="ui basic icon button"
                  onClick={[Function]}
                >
                  <svg
                    className="icon"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": undefined,
                      }
                    }
                    onClick={null}
                    style={
                      Object {
                        "fill": "currentColor",
                        "height": "24px",
                        "width": "auto",
                      }
                    }
                    viewBox=""
                    xmlns=""
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
